MCprep_addon/commonmcobj_parser.py

```diff
@@ -214,34 +214,38 @@ def clean_and_extract(line: str) -> Tuple[str, str]:
 
     return header
 
+
 def parse_header(f: TextIO) -> Optional[CommonMCOBJ]:
     """
-    Parses a file and returns a CommonMCOBJ object if 
-    the header exists.
-    
+    Parses a file and returns a CommonMCOBJ object if the header exists.
+
     f: TextIO
         File object
-    
+
     Returns:
         - CommonMCOBJ object if header exists
         - None otherwise
     """
 
     header: List[str] = []
     found_header = False
-    
+
     # Read in the header
-    for l in f:
-        tl = " ".join(l.rstrip().split())
-        if tl == "# COMMON_MC_OBJ_START":
+    lines_read = 0
+    for _l in f:
+        tl = " ".join(_l.rstrip().split())
+        lines_read += 1
+        if lines_read > 100 and tl and not tl.startswith("#"):
```

Comment:
I'll see if I can make this limit configurable in user preferences

Reply:
I'd prefer we keep things simple personally, adding preferences for everything makes the preferences section long and hard to read if every minute thing becomes an option (even if it's pushed under advanced). I'm not particularly worried about running into this problem.

If anything, let's double it to 200, how does that sound? For smaller obj we're still typically 1000's of lines of the overall file so it's still a big parsing improvement over reading the while thing (twice)

Reply:
200 sounds like a good compromise. If an OBJ doesn't have the header within the first 200 lines, we can resolve it at that time (if it happens)